ipaserver/plugins/certmap.py

```diff
@@ -91,6 +91,53 @@
 register = Registry()
 
 
+def check_maprule_is_for_trusted_domain(api_inst, options, entry_attrs):
+    """
+    Check that the certmap rule references altSecurityIdentities and
+    associateddomain in options is a trusted domain.
+
+    :param api_inst: API instance
+    :param options: options passed to the command
+                    at least ipacertmapmaprule and
+                    associateddomain options are expected for the check
+
+    :raises: ValidationError if altSecurityIdentities is present in the
+             rule but no Active Directory trusted domain listed in
+             associated domains
+    """
+    is_trusted_domain_required = False
+    # If no explicit option passed, fallback to the content of the entry
+    # This helps to catch cases when an associated domain value is removed
+    # while the rule requires its presence
+    maprule = options.get('ipacertmapmaprule',
+                          entry_attrs.get('ipacertmapmaprule'))
+    if maprule:
+        if 'altsecurityidentities' in maprule.lower():
+            is_trusted_domain_required = True
+
+    if is_trusted_domain_required:
+        domains = options.get('associateddomain',
+                              entry_attrs.get('associateddomain'))
+        if domains:
+            trusted_domains = api_inst.Object.config.gather_trusted_domains()
+            trust_suffix_namespace = {dom_name.lower() for dom_name in
+                                      trusted_domains}
+            for dom in domains:
```

Comment:
The check should rather be that at least one of the domains is in trust_suffix_namespace. For instance, if `--domain ipa.domain.com --domain ad.domain.com` are specified, a maprule with altsecurityIdentities should be accepted (as in the example 23.4 of [Examples of Identity Mapping Rules](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/linux_domain_identity_authentication_and_policy_guide/smart-cards#sc-id-mapping-rule-examples) where a single rule is used for all the domains).

Reply:
@flo-renaud Thanks. I think new code covers it.